src/main/java/uk/gov/pay/directdebit/mandate/dao/MandateSearchDao.java

```diff
@@ -7,159 +7,98 @@
-    private static final String MANDATE_CREATED_DATE_COLUMN_NAME = "created_date";
-    
-    private static final String PAYER_NAME_COLUMN_NAME = "name";
-    private static final String PAYER_EMAIL_COLUMN_NAME = "email";
-
     private final Jdbi jdbi;
 
-    private final String QUERY_STRING = "SELECT DISTINCT" +
-            "  m.id AS mandate_id," +
-            "  m.external_id AS mandate_external_id," +
-            format("  m.%s AS mandate_mandate_reference,", MANDATE_REFERENCE_COLUMN_NAME) +
-            format("  m.%s AS mandate_service_reference,", MANDATE_SERVICE_REFERENCE_COLUMN_NAME) +
-            "  m.gateway_account_id AS mandate_gateway_account_id," +
-            "  m.return_url AS mandate_return_url," +
-            format("  m.%s AS mandate_state,", MANDATE_STATE_COLUMN_NAME) +
-            format("  m.%s AS mandate_created_date,", MANDATE_CREATED_DATE_COLUMN_NAME) +
-            "  m.payment_provider_id AS mandate_payment_provider_id," +
-            "  g.id AS gateway_account_id," +
-            "  g.external_id AS gateway_account_external_id," +
-            "  g.payment_provider AS gateway_account_payment_provider," +
-            "  g.type AS gateway_account_type," +
-            "  g.description AS gateway_account_description," +
-            "  g.analytics_id AS gateway_account_analytics_id," +
-            "  g.access_token AS gateway_account_access_token," +
-            "  g.organisation AS gateway_account_organisation," +
-            "  p.id AS payer_id," +
-            "  p.mandate_id AS payer_mandate_id," +
-            "  p.external_id AS payer_external_id," +
-            format("  p.%s AS payer_name,", PAYER_NAME_COLUMN_NAME) +
-            format("  p.%s AS payer_email,", PAYER_EMAIL_COLUMN_NAME) +
-            "  p.bank_account_number_last_two_digits AS payer_bank_account_number_last_two_digits," +
-            "  p.bank_account_requires_authorisation AS payer_bank_account_requires_authorisation," +
-            "  p.bank_account_number AS payer_bank_account_number," +
-            "  p.bank_account_sort_code AS payer_bank_account_sort_code," +
-            "  p.bank_name AS payer_bank_name," +
-            "  p.created_date AS payer_created_date" +
-            " FROM mandates m" +
-            format("  JOIN gateway_accounts g ON g.id = m.%s ", MANDATE_GATEWAY_ACCOUNT_ID_COLUMN_NAME) +
-            "  LEFT JOIN payers p ON p.mandate_id = m.id " +
-            "WHERE g.external_id = :gatewayAccountExternalId " +
-            "  :searchExtraFields "; 
-//            "  ORDER BY p.id DESC OFFSET :offset LIMIT :limit";
-
     @Inject
     public MandateSearchDao(Jdbi jdbi) {
         this.jdbi = jdbi;
     }
 
     public List<Mandate> search(MandateSearchParams mandateSearchParams) {
-        Params params = new Params(mandateSearchParams);
-        String searchExtraFields = generateQuery(params);
+        var sql = new StringBuilder();
```

Comment:
The default capacity of a `StringBuilder` is 16 and I think it roughly doubles in size every time it reaches the limit. Since this string is always going to be quite large, we’re going to have a few resizes. Can we do `new StringBuilder(2048)` instead?